app/models/manageiq/providers/lenovo/physical_infra_manager/event_parser.rb

```diff
@@ -32,7 +36,29 @@ def self.filter_data(data)
     }
   end
 
-  def self.get_physical_server_id(ems_ref)
-    PhysicalServer.find_by(:ems_ref => ems_ref).try(:id)
+  class << self
+    private
+
+    def event_resources(event)
+      event_resources = {}
+
+      if event[:parent_type] == CHASSIS
+        event_resources[:physical_chassis_id] = get_klass_id(PhysicalChassis, event[:parent_uuid])
+      end
+
+      if event[:component_type] == SWITCH
+        event_resources[:physical_switch_id] = get_klass_id(PhysicalSwitch, event[:component_id])
+      elsif SERVER.include?(event[:component_type])
+        event_resources[:physical_server_id] = get_klass_id(PhysicalServer, event[:component_id])
+      else
+        raise(NotImplementedError, "The event of type #{event[:component_type]} is not supported")
```

Comment:
I think that is a little better only create a new log output like `$log.erro(message)` instead of raise an exception. @felipedf do you agree ?